utils/check_clang_tidy_results.py

```diff
@@ -0,0 +1,161 @@
+    '[hicpp-signed-bitwise]',
+    '[hicpp-special-member-functions]',
+    '[hicpp-use-auto]',
+    '[hicpp-use-emplace]',
+    '[hicpp-use-equals-default]',
+    '[hicpp-use-nullptr]',
+    '[hicpp-use-override]',
+    '[hicpp-vararg]',
+    '[llvm-namespace-comment]',
+    '[misc-macro-parentheses]',
+    '[misc-misplaced-widening-cast]',
+    '[misc-redundant-expression]',
+    '[misc-string-integer-assignment]',
+    '[misc-suspicious-enum-usage]',
+    '[misc-suspicious-string-compare]',
+    '[misc-uniqueptr-reset-release]',
+    '[misc-unused-using-decls]',
+    '[modernize-loop-convert]',
+    '[modernize-make-shared]',
+    '[modernize-make-unique]',
+    '[modernize-pass-by-value]',
+    '[modernize-raw-string-literal]',
+    '[modernize-return-braced-init-list]',
+    '[modernize-use-auto]',
+    '[modernize-use-bool-literals]',
+    '[modernize-use-default-member-init]',
+    '[modernize-use-emplace]',
+    '[modernize-use-equals-default]',
+    '[modernize-use-nullptr]',
+    '[modernize-use-override]',
+    '[performance-for-range-copy]',
+    '[performance-inefficient-string-concatenation]',
+    '[performance-inefficient-vector-operation]',
+    '[performance-unnecessary-copy-initialization]',
+    '[performance-unnecessary-value-param]',
+    '[readability-container-size-empty]',
+    '[readability-delete-null-pointer]',
+    '[readability-else-after-return]',
+    '[readability-implicit-bool-conversion]',
+    '[readability-inconsistent-declaration-parameter-name]',
+    '[readability-named-parameter]',
+    '[readability-non-const-parameter]',
+    '[readability-redundant-control-flow]',
+    '[readability-redundant-member-init]',
+    '[readability-redundant-smartptr-get]',
+    '[readability-redundant-string-cstr]',
+    '[readability-redundant-string-init]',
+    '[readability-simplify-boolean-expr]',
+    '[readability-static-accessed-through-instance]',
+    '[readability-static-definition-in-anonymous-namespace]'
+}
+
+CHECK_REGEX = re.compile('.*\[([A-Za-z0-9.-]+)\]$')
+
+
+def main():
+    """Checks whether clang-tidy warnings that were previously cleaned have
+    regressed."""
+    if len(sys.argv) == 2:
+        print('#######################################')
+        print('#######################################')
+        print('###                                 ###')
+        print('###   Checking clang-tidy results   ###')
+        print('###                                 ###')
+        print('#######################################')
+        print('#######################################')
+    else:
+        print(
+            'Usage: check_clang_tidy_results.py <log_file>')
+        return 1
+
+    log_file = sys.argv[1]
+
+    errors = 0
+
+    with open(log_file) as checkme:
+        contents = checkme.readlines()
+        for line in contents:
+            if 'third_party' in line:
+                continue
+            check_suppressed = False
+            for check in SUPPRESSED_CHECKS:
```

Comment:
This SUPPRESSED_CHECKS design looks rather fragile to me. Wouldn't it be better to instead filter for a brief list of checks which we want?

Reply:
That's is the design I had at first, but this way around it is a lot more maintainable,e.g. when we want to clear more checks, we already have a ready-made list of checks that are available.